url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/net/http/task/server/HttpFrontWaitParseableHeader.java

```diff
@@ -0,0 +1,36 @@
+package net.hetimatan.net.http.task.server;
+
+
+import java.lang.ref.WeakReference;
+
+import net.hetimatan.net.http.HttpServerFront;
+import net.hetimatan.util.event.EventTask;
+import net.hetimatan.util.event.EventTaskRunner;
+
+public class HttpFrontWaitParseableHeader extends EventTask {
+	public static final String TAG = "HttpFrontWaitParseableHeader";
+	private WeakReference<HttpServerFront> mClientInfo = null;
+
+	public HttpFrontWaitParseableHeader(HttpServerFront clientInfo) {
+		mClientInfo = new WeakReference<HttpServerFront>(clientInfo);
+		errorAction(new HttpFrontCloseTask(clientInfo));
+	}
+
+	@Override
+	public String toString() {
+		return TAG;
+	}
+
+	@Override
+	public void action(EventTaskRunner runner) throws Throwable {
+		HttpServerFront info = mClientInfo.get();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *info*. Execution paths that do not contain closure statements were detected.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).